InvenTree/label/api.py

```diff
@@ -136,17 +138,47 @@ def filter_queryset(self, queryset):
 class LabelPrintMixin(LabelFilterMixin):
     """Mixin for printing labels."""
 
+    rolemap = {
+        "GET": "view",
+        "POST": "view",
+    }
+
+    def check_permissions(self, request):
+        """Override request method to GET so that also non superusers can print using a post request."""
+        if request.method == "POST":
+            request = clone_request(request, "GET")
+        return super().check_permissions(request)
```

Comment:
No, I wanted to get the read permission for the specific Label type (part, stock, line, ...) to work and don't just handle all the same.

Reply:
I think the comment above and this comment correlate to each other. I'm open for any other solution, but this was the best method I could think of. I needed both adjustments in order for the metadata generation to work, because the SimpleMetadata from DRF internally only uses this check_permissions function, and the Inventree metadata class filters out any methods that the user is not allowed to access by using the roleset.

Reply:
Now that you have a custom rolemap defined, do you still need the custom `check_permission` method here?

Reply:
@SchrodingersGat Yes, that's was what I was trying to explain in the comment above.

Reply:
Ah, gotcha

Reply:
Couldn't we just directly check for the read/view permission in this function?

Reply:
@matmair Yeah, I could duplicate the code if you think that's a better approach, but that was what I was trying to avoid

Reply:
let's go with this for now, I will put it onto my refactoring todo